types/src/blob.rs

```diff
@@ -68,66 +68,65 @@ pub struct BlobParams {
 }
 
 impl Blob {
-    /// Create a new blob with the given data within the [`Namespace`].
+    /// Create a new blob with the given data within the [`Namespace`], with optional signer.
+    ///
+    /// # Notes
+    ///
+    /// If present onchain, `signer` was verified by consensus node on blob submission.
     ///
     /// # Errors
     ///
     /// This function propagates any error from the [`Commitment`] creation.
+    /// To use `signer = Some(address)`, [`AppVersion`] must be at least [`AppVersion::V3`].
     ///
     /// # Example
     ///
     /// ```
-    /// use celestia_types::{AppVersion, Blob, nmt::Namespace};
+    /// use celestia_types::{AppVersion, Blob, nmt::Namespace, state::AccAddress};
     ///
     /// let my_namespace = Namespace::new_v0(&[1, 2, 3, 4, 5]).expect("Invalid namespace");
-    /// let blob = Blob::new(my_namespace, b"some data to store on blockchain".to_vec(), AppVersion::V2)
-    ///     .expect("Failed to create a blob");
+    /// let signer: AccAddress = "celestia1377k5an3f94v6wyaceu0cf4nq6gk2jtpc46g7h"
+    ///     .parse()
+    ///     .unwrap();
+    /// let blob_signed = Blob::new(
+    ///     my_namespace,
+    ///     b"some data to store on blockchain".to_vec(),
+    ///     Some(signer),
+    ///     AppVersion::V5,
+    /// )
+    /// .expect("Failed to create a signed blob");
+    ///
+    /// let actual_signed = serde_json::to_string_pretty(&blob_signed).unwrap();
+    /// let expected_signed = indoc::indoc! {r#"{
+    ///   "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQIDBAU=",
+    ///   "data": "c29tZSBkYXRhIHRvIHN0b3JlIG9uIGJsb2NrY2hhaW4=",
+    ///   "share_version": 1,
+    ///   "commitment": "AUpLKHYnlrfK0cX6DcGyryFMld1nJia+cjkCwXFTFgA=",
+    ///   "index": -1,
+    ///   "signer": "j71qdnFJas04ncZ4/CazBpFlSWE="
+    /// }"#};
     ///
     /// assert_eq!(
-    ///     &serde_json::to_string_pretty(&blob).unwrap(),
-    ///     indoc::indoc! {r#"{
-    ///       "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQIDBAU=",
-    ///       "data": "c29tZSBkYXRhIHRvIHN0b3JlIG9uIGJsb2NrY2hhaW4=",
-    ///       "share_version": 0,
-    ///       "commitment": "m0A4feU6Fqd5Zy9td3M7lntG8A3PKqe6YdugmAsWz28=",
-    ///       "index": -1,
-    ///       "signer": null
-    ///     }"#},
+    ///     serde_json::from_str::<serde_json::Value>(&actual_signed).unwrap(),
+    ///     serde_json::from_str::<serde_json::Value>(expected_signed).unwrap()
```

Comment:
could we maybe restore the original assert?

```rust
assert_eq!(
    serde_json::to_string_pretty(&blob).unwrap(),
    indoc::indoc! {r#"{
      "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQIDBAU=",
      "data": "c29tZSBkYXRhIHRvIHN0b3JlIG9uIGJsb2NrY2hhaW4=",
      "share_version": 1,
      "commitment": "AUpLKHYnlrfK0cX6DcGyryFMld1nJia+cjkCwXFTFgA=",
      "index": -1,
      "signer": "j71qdnFJas04ncZ4/CazBpFlSWE="
    }"#},
);
```

I like it for readability, going back and forth through serde just adds noise, especially that we compare `Value`. In case there's some issue with that (idk, field order during serialization :thinking:) then I would prefer comparing `Blob`s with a single call to `from_str` on the hardcoded json